rayon-core/src/scope/mod.rs

```diff
@@ -278,7 +285,7 @@ impl<'scope> Scope<'scope> {
 
             // the `Scope` is not send or sync, and we only give out
             // pointers to it from within a worker thread
-            debug_assert!(!WorkerThread::current().is_null());
+            debug_assert!(!worker_thread.is_null());
 
             let worker_thread = &*worker_thread;
             worker_thread.push(job_ref);
```

Comment:
Maybe this can just switch to `scope.registry.inject_or_push(job_ref)`, so if the scope does get carried out of the threadpool, it still spawns back to the right pool.  The `'scope` lifetime should still ensure the right stuff...

Reply:
Done.